### ISSUE TITLE
Class renaming is broken when you have other tab opened with import

### ISSUE THREAD

**Issue by brndt** [scalameta/metals#6478]
### Describe the bug

1. Rename class A name to B
2. Everything works
3. Open class C that have imported class B
4. Try to rename class B to class D
5. Sometimes is broken as you can see in the video
https://github.com/scalameta/metals/assets/21007364/49fe71de-7583-4ee9-b2a1-0cc7aaffa339

Maybe related to: https://github.com/scalameta/metals/issues/5759

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v1.3.1

**Comment by tgodzik** [comment]
Getting back to it think the issue here is that sometimes the newly renamed class is not compiled and I see old classfiles together with new semanticdb, which contains old information.

Not sure yet if the problem is Bloop or Zinc. But the workaround is to force the compilation for the file again by adding some lines etc.

**Comment by tgodzik** [comment]
Looks like https://github.com/scalacenter/bloop/pull/2490 could help from my testing.

**Comment by brndt** [comment]
@tgodzik thanks for your reply!
How can I test it? As far as I remember you can't change Bloop version in the VS Code settings.

**Comment by brndt** [comment]
Also, I have a question: Does renaming a class produce the same effect as moving it to another folder? Because I have the same experience for both cases, if I make one or another change (renaming or repackaging) in the `main` folder, only the references to this class from the `test` are starting to break. That's why I think it could be related to this [issue]. (https://github.com/scalameta/metals/issues/5759)

**Comment by tgodzik** [comment]
> @tgodzik thanks for your reply! How can I test it by myself?

You would need to checkout my branch, do `sbt publishLocal`, change `metals.bloopVersion` in metals settings and restart the build server (you should be prompted automatically)

> Also, I have a question: Does renaming a class produce the same effect as moving it to another folder? Because I have the same experience for both cases, if I make one or another change (renaming or repackaging) in the main folder, only the references to this class from the test are starting to break. That's why I think it could be related to this [issue]. (https://github.com/scalameta/metals/issues/5759)

I think so, but might be worth checking out. If we try to compile a file that no longer exist then something weird might happen. My hope is that this is the cause of many weird metals bugs, but I will need to see.

**Comment by brndt** [comment]
https://github.com/user-attachments/assets/4f4fb70a-0c84-40ce-8979-8e3d86d546ab

I didn't work for me :(

You can reproduce my case by opening CreateArticle and CreateArticleSpec classes and then renaming (o moving to another folder) CreateArticle class. I gave you an access to `spursarmy-main` repo.

**Comment by tgodzik** [comment]
That looks like another issue than the one I found. It would stop renaming altogether for me after a couple of renames, here it looks like it looses a couple of references. I might have misunderstood the issue in your first video then. I will take a look at what you posted now.

**Comment by brndt** [comment]
Yeah, sometimes it happens too - class is renamed, bad badly recompiled. But normally it's not renamed and badly recompiled.
I guess maybe if you try it to rename it several times (or try different class from the `main`) you'll get my case too.

**Comment by tgodzik** [comment]
The issue is somewhere in Bloop, so we should not work around that. I wasn't able to reproduce this after recent fixes to Bloop, but I will get back to it and try again.

**Comment by brndt** [comment]
I checked it again and seems like the renaming is still broken sometimes, I don't what's changing but there are times when it's working fine but then it suddenly stops working.

**Comment by brndt** [comment]
The strange thing is that renaming is broken only in `test` folder, all references from `main` are renamed correctly always, just as I mentioned before.

> Also, I have a question: Does renaming a class produce the same effect as moving it to another folder? Because I have the same experience for both cases, if I make one or another change (renaming or repackaging) in the `main` folder, only the references to this class from the `test` are starting to break. That's why I think it could be related to this [issue]. ([#5759](https://github.com/scalameta/metals/issues/5759))

**Comment by tgodzik** [comment]
Och, that's curious, will take a look.
